### ISSUE TITLE
Typo in example

### ISSUE THREAD

**Issue by NOhs** [Trick-17/clang-build#15]
In the readme it says

```toml
# Include an external package/target (i.e. not from this toml file)
[somelib]
external = true
path = "/path/to/sources"

# Build an executable and link the library
[myexe]
[myexe.sources]
include_directories = ["include", "mylib.sources.include_directories"]
source_directories = ["src"]

[myexe.link]
dependencies = ["somelib"]
```

Shouldn't it read: `somelib.sources.include_directories`?

Also I am not sure I understand the example. So we have the linker dependency on `somelib`. Why do we have to explicitly give it the source files again? This seems redundant and error prone (one might forget one or the other and not directly notice it, because there is an entry already that states that there is a dependency).

**Comment by GPMueller** [comment]
What do you mean by "again"? The target `somelib` has to be defined somehow, so one should either specify a root folder for the target (as in this example) or a git repo address, or not define it as an `external` project. The build system needs to know where to look for either a build config file or the sources.
*If* dependency management would be added, it would probably be a different story.

However, I'm not sure that `external` makes much sense anymore, since such targets could be treated just the same as regular ones... But that's maybe a separate discussion.

It may be useful to be able to use a different projects include_directories, but they shouldn't be forwarded further up (I suppose), so it should be enough to do it implicitly (i.e. if a target1 depends on target2, the include_directories of target2 should be automatically added to those of target1).

**Comment by NOhs** [comment]
I meant the `include_directories = ["include", "mylib.sources.include_directories"]`. Is that a typo? `mylib=somelib`? If so, I don't see the reason for adding the include directory, given the fact that we know that this executable depends on the library, we should include the relevant sources *automatically*.

**Comment by GPMueller** [comment]
You're right. However, if there is a dependency chain, the include directories should only be passed to the direct parents, not further up the chain right?
```
A  - does not see the include directories of C
| A depends on B
B  - include_directories include those of C
| B depends on C
C  - defines some include directories
```

**Comment by NOhs** [comment]
Hm, currently I have it all the way to the end. Because I don't know if there is a safe way to decide when to stop. But this shouldn't be an issue. Seeing that a lot of software installed on linux puts some headers into the default include directories, this should not be a performance issue at least.

**Comment by GPMueller** [comment]
I cannot think of a case where this would be an issue, but I would imagine that there is some reason that CMake has this feature of "interface" includes/libraries etc. (maybe defining two kinds of include directories would be the right way in our case, too?).
Maybe we should research this a little, but for now it will probably be fine if includes are infinitely forwarded.

Note that this may pollute the compile/link command of the highest target, which could in turn make the verbose output completely unreadable. If there is a sensible rule for when to stop I would like to have it implemented.

**Comment by GPMueller** [comment]
For now, this is resolved since the merge of PR #23 with a7bbc88738d7863cfe6f0994f80914a99914d417.
